src/Microsoft.OData.Edm/Csdl/Semantics/CsdlSemanticsModel.cs

```diff
@@ -574,34 +600,50 @@ private void AddSchema(CsdlSchema schema, bool addAnnotations)
             {
                 RegisterElement(container);
             }
+        }
 
-            if (!string.IsNullOrEmpty(schema.Alias))
+        private void AddOutOfLineAnnotationsFromSchema(CsdlSchema schema, CsdlSemanticsSchema schemaWrapper, IEnumerable<IEdmIncludeAnnotations> includeAnnotations)
+        {
+            foreach (CsdlAnnotations schemaOutOfLineAnnotations in schema.OutOfLineAnnotations)
             {
-                this.SetNamespaceAlias(schema.Namespace, schema.Alias);
-            }
+                string target = this.ReplaceAlias(schemaOutOfLineAnnotations.Target);
 
-            if (addAnnotations)
-            {
-                foreach (CsdlAnnotations schemaOutOfLineAnnotations in schema.OutOfLineAnnotations)
+                List<CsdlSemanticsAnnotations> annotations;
+                if (!this.outOfLineAnnotations.TryGetValue(target, out annotations))
                 {
-                    string target = this.ReplaceAlias(schemaOutOfLineAnnotations.Target);
-
-                    List<CsdlSemanticsAnnotations> annotations;
-                    if (!this.outOfLineAnnotations.TryGetValue(target, out annotations))
-                    {
-                        annotations = new List<CsdlSemanticsAnnotations>();
-                        this.outOfLineAnnotations[target] = annotations;
-                    }
-
-                    annotations.Add(new CsdlSemanticsAnnotations(schemaWrapper, schemaOutOfLineAnnotations));
+                    annotations = new List<CsdlSemanticsAnnotations>();
+                    this.outOfLineAnnotations[target] = annotations;
                 }
+
+                CsdlAnnotations filteredAnnotations = FilterIncludedAnnotations(schemaOutOfLineAnnotations, target, includeAnnotations);
+                annotations.Add(new CsdlSemanticsAnnotations(schemaWrapper, filteredAnnotations));
             }
+        }
 
-            var edmVersion = this.GetEdmVersion();
-            if (edmVersion == null || edmVersion < schema.Version)
+        private CsdlAnnotations FilterIncludedAnnotations(CsdlAnnotations csdlAnnotations, string target, IEnumerable<IEdmIncludeAnnotations> includeAnnotations)
+        {
+            if (includeAnnotations == null || !includeAnnotations.Any())
             {
-                this.SetEdmVersion(schema.Version);
+                return csdlAnnotations;
             }
+
+            var filteredAnnotations = csdlAnnotations.Annotations.Where(annotation =>
+                ShouldIncludeAnnotation(annotation, target, includeAnnotations));
+            return new CsdlAnnotations(filteredAnnotations, csdlAnnotations.Target, csdlAnnotations.Qualifier);

```

Comment:
>csdlAnnotations.Target [](start = 60, length = 22)

should this use csdlAnnotations.Target, or "target" (which has the alias already resolved)?

Reply:
The aim here is to return a copy of the `CsdlAnnotations` instance but with the actual annotations filtered. So every other property should be identical to the original `CsdlAnnotations` instance, hence `csdlAnnotations.Target`. The `target` with the alias already resolved is used when comparing against the `include.TargetNamespace`.